paper-linear-progress.html

```diff
@@ -0,0 +1,897 @@
+          this._playCorrectBufferingAnimation();
+        }
+      },
+
+      /**
+       * The `paper-linear-progress-value-change` event is fired whenever the progress value
+       * changes.
+       * @event paper-linear-progress-value-change
+       * @detail {{value: Number}}
+       */
+
+      _valueChanged: function(newValue) {
+        this._scaleToPercent(newValue, this.$.progressPrimary);
+        if (!this.indeterminate) {
+          this.setAttribute('aria-valuenow', newValue);
+        }
+        this.fire('paper-linear-progress-value-change', {value: newValue});
+      },
+
+      /**
+       * The `paper-linear-progress-buffer-value-change` event is fired whenever the buffer value
+       * changes.
+       * @event paper-linear-progress-buffer-value-change
+       * @detail {{bufferValue: Number}}
+       */
+
+      _bufferValueChanged: function(newBufferValue) {
+        this._scaleToPercent(newBufferValue, this.$.buffer);
+        this.fire('paper-linear-progress-buffer-value-change', {bufferValue: newBufferValue});
+      },
+
+      _indeterminateChanged: function(isIndeterminate) {
+        this._forceLayoutIfHasClass('reversed');
+        this.toggleClass('indeterminate', isIndeterminate);
+        if (isIndeterminate) {
+          this.removeAttribute('aria-valuemin');
+          this.removeAttribute('aria-valuemax');
+          this.removeAttribute('aria-valuenow');
+        } else {
+          this.setAttribute('aria-valuemin', 0);
+          this.setAttribute('aria-valuemax', 100);
+          this.setAttribute('aria-valuenow', this.value);
+        }
+      },
+
+      _reversedChanged: function(isReversed) {
+        this._forceLayoutIfHasClass('indeterminate');
+        this.toggleClass('reversed', isReversed);
+        if (this.buffering) {
+          this._playCorrectBufferingAnimation();
+        }
+      },
+
+      _bufferingChanged: function(isBuffering) {
+        this.toggleClass('buffering', isBuffering);
+
+        var isAnimationReady = 'buffering' in this.animationConfig;
+        if (isBuffering) {
+          if (isAnimationReady) {
+            this._playCorrectBufferingAnimation();
+          } else {
+            this.__needsBufferingAnimationOnAttached = true;
+          }
+        } else {
+          if (isAnimationReady) {
+            this.cancelAnimation();
+          } else {
+            this.__needsBufferingAnimationOnAttached = false;
+          }
+        }
+      },
+
+      _closedChanged: function(isClosed) {
+        this.toggleClass('closed', isClosed);
+      },
+
+      _applyBufferingDotsBackground: function() {
+        var bgColor = window.getComputedStyle(this.$.buffer).backgroundColor;
+        var prologue = 'url("data:image/svg+xml;charset=UTF-8,';
+        // NOTE(traviskaufman): Single-quotes are needed around SVG attributes in order for it to
+        // render properly in all browsers, hence the escaped single-quote strings as opposed to
+        // double-quoting.
```

Comment:
Also, did you consider using the `Blob` / `URL.createObjectURL` approach? This might yield different results with regards to parsing quotes.

Also x 2: It might be nice to make this URL once for all `paper-linear-progress` elements, and then re-use it. Do you think that's possible? The variable background color might make this tricky but... maybe there is a way?